lib/manageiq/messaging/kafka/client.rb

```diff
@@ -53,27 +43,27 @@ class Client < ManageIQ::Messaging::Client
         include BackgroundJob
         include Topic
 
-        private *delegate(:subscribe, :unsubscribe, :publish, :to => :kafka_client)
-        delegate :close, :to => :kafka_client
-
         attr_accessor :encoding
 
         def ack(ack_ref)
-          @queue_consumer.try(:mark_message_as_processed, ack_ref)
-          @topic_consumer.try(:mark_message_as_processed, ack_ref)
+          ack_ref.commit
+        rescue Rdkafka::RdkafkaError => e
+          logger.warn("ack failed with error #{e.message}")
+          raise unless e.message =~ /no_offset/
         end
 
         def close
-          @topic_consumer.try(:stop)
-          @topic_consumer = nil
-          @queue_consumer.try(:stop)
-          @queue_consumer = nil
-
-          @producer.try(:shutdown)
+          @producer&.close
           @producer = nil
 
-          kafka_client.close
-          @kafka_client = nil
+          @consumer&.close
+          @consumer = nil
+        end
+
+        # list all topics
+        def topics
+          native_kafka = producer.instance_variable_get(:@native_kafka)
+          Rdkafka::Metadata.new(native_kafka).topics.collect { |topic| topic[:topic_name] }
```

Comment:
If the upstream is responsive maybe you could open a PR to add a `Rdkafka::Producer#list_topics` method so we don't have to `instance_variable_get` ?

Reply:
Yes we could. I add this only because sources-api uses this feature now.

Reply:
Yeah it seems like a great feature to have